### PR TITLE
handle libfuzzer targets failing without dropping crash files

### PR DESCRIPTION
**Pull request by bmc-msft**
## Summary of the Pull Request

Handle libfuzzer targets exiting with an error without recording crash files

## PR Checklist
* [X] Applies to work item: #57 
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [X] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

* Moves `src/process.rs` from `onefuzz-supervisor` to `onefuzz`
* Keeps a copy of stderr in case the fuzzer crashes without dropping crash file
* Adds `onefuzz-agent debug libfuzzer_fuzz` debug subcommand to test libfuzzer targets locally

## Validation Steps Performed

* Deploy a regular libfuzzer task and note it detects crashes as appropriate 
* Deploy a target that does `raise(SIGKILL);` and note that we mark the task is failed

### CODE REVIEW

File: src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs [hunk 1]
```diff
@@ -139,27 +141,43 @@ impl LibFuzzerFuzzTask {
...
+        let files = list_files(crash_dir.path()).await?;
 
-                    if let Err(err) = try_report_iter_update(stats_sender, worker_id, run_id, line)
-                    {
-                        error!("could not parse fuzzing iteration update: {}", err);
-                    }
-                }
-            })
-            .await;
+        // ignore libfuzzer exiting cleanly without crashing, which could happen via
```
Comment (ranweiler) [inline src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs:163]: 👍

File: src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs [hunk 2]
```diff
@@ -139,27 +141,43 @@ impl LibFuzzerFuzzTask {
             .stderr
             .as_mut()
             .ok_or_else(|| format_err!("stderr not captured"))?;
-        let stderr = io::BufReader::new(stderr);
+        let stderr = BufReader::new(stderr);
+
+        let mut libfuzzer_output = Vec::new();
+        let mut lines = stderr.lines();
+        while let Some(line) = lines.next_line().await? {
+            // let stats_sender = stats_sender
```
Comment (ranweiler) [inline src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs:149]: Stray commented code.

File: src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs [hunk 3]
```diff
@@ -35,7 +39,7 @@ const RESYNC_PERIOD: Duration = Duration::from_secs(30);
 const PROC_INFO_COLLECTION_DELAY: Duration = Duration::from_secs(1);
 
 // Period of reporting proc info about running processes.
-const PROC_INFO_PERIOD: Duration = Duration::from_secs(30);
+const PROC_INFO_PERIOD: Duration = Duration::from_secs(10);
```
Comment (ranweiler) [inline src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs:42]: Is this change related?

**Review by ranweiler — APPROVED** [review]

File: src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs (hunk 3 above)
Comment (bmc-msft) [reply src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs]: Debugging artifact.  How we collect processor stats for libfuzzer targets is timer based, which was frustrating during debugging of this issue.  I've reset this, but we should address how processor stats are collected in the future.
